packages/eas-cli/src/project/projectUtils.ts

```diff
@@ -52,13 +60,56 @@ export async function findProjectRootAsync(cwd?: string): Promise<string | null>
   return projectRootDir ?? null;
 }
 
-export async function getProjectIdAsync(exp: ExpoConfig): Promise<string> {
+export async function setProjectIdAsync(projectDir: string): Promise<ExpoConfig | undefined> {
+  const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+
   const privacy = toAppPrivacy(exp.privacy);
-  return await ensureProjectExistsAsync({
+  const projectId = await ensureProjectExistsAsync({
     accountName: getProjectAccountName(exp, await ensureLoggedInAsync()),
     projectName: exp.slug,
     privacy,
   });
+
+  const result = await modifyConfigAsync(projectDir, {
+    extra: { ...exp.extra, eas: { ...exp.extra?.eas, projectId } },
+  });
+
+  if (result.type !== 'success') {
+    throw new Error(result.message);
```

Comment:
maybe in the case of dynamic config, we should display instructions for the user, that either 
- suggest using both static and dynamic configs and rerurning command
- prints value of project id and location where should be in config

Reply:
* Even if we are using both, the method still throws: https://github.com/expo/expo-cli/blob/master/packages/config/src/Config.ts#L383
And perhaps this is the write behavior as we don't know if they would want this in the static or dynamic config.

* I added the following message and retested various other commands with a dynamic setup.
<img width="1140" alt="Screen Shot 2021-05-12 at 7 32 23 AM" src="https://user-images.githubusercontent.com/1220444/117993011-32dc6100-b2f4-11eb-8b1c-0e145824dbef.png">